modules/analyse/src/main/Analysis.scala

```diff
@@ -30,15 +28,41 @@ case class Analysis(
       })
     }
 
-  def valid = infos.nonEmpty
+  def studyId = id.studyId
+  def valid   = infos.nonEmpty
 
   def nbEmptyInfos       = infos.count(_.isEmpty)
   def emptyRatio: Double = nbEmptyInfos.toDouble / infos.size
 
 object Analysis:
 
-  opaque type Id = String
-  object Id extends OpaqueString[Id]
+  enum Id:
+    case Game(id: GameId)
+    case Study(study: StudyId, id: StudyChapterId)
+
+  object Id:
+    def apply(gameId: GameId): Id                          = Game(gameId)
+    def apply(study: StudyId, chapter: StudyChapterId): Id = Study(study, chapter)
+    def apply(study: Option[StudyId], id: String): Id =
+      study.fold(Game(GameId(id)))(Study(_, StudyChapterId(id)))
+
+    extension (id: Id)
+
+      def id: String = id match
```

Comment:
Bike-shedding but maybe ‘value’ would be more consistent name?

Reply:
I thought about it, but I didn't have a strong opinion. But I think `value` is better here for consistency as you mentioned.